src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorPage.cs

```diff
@@ -269,6 +272,50 @@ public new bool IsSectionDefined(string name)
             return zone != null && zone.Items.Count > 0;
         }
 
+        public new void DefineSection(string name, Microsoft.AspNetCore.Mvc.Razor.RenderAsyncDelegate section)
+        {
+            var zone = ThemeLayout.Zones[name];
+            dynamic sectionShape = New.AspSection().GetAwaiter().GetResult();
+            sectionShape.RenderAsyncDelegate = SectionDelegate(name, section, sectionShape);
+
+            if (zone is Zones.ZoneOnDemand zoneOnDemand)
+            {
+                zoneOnDemand.AddAsync(sectionShape).GetAwaiter().GetResult();
+            }
```

Comment:
I don't think there is a way around it, but this `.GetAwaiter().GetResult()` is a bit of a sticking point for performance reasons.

Reply:
I don't think so as `DefineSection` is not async

That said this one is only called if the related zone is not already there, if in the view you use the zone taghelper (liquid / razor) for the targetted zone, this is the line 287 below that is called.

Then, to not use `New.AspSection().GetAwaiter().GetResult()`, in this case i suggest the following without using the shape factory and without first working on a dynamic shape

            var zone = ThemeLayout.Zones[name];

            var sectionShape = new Shape();
            sectionShape.Metadata.Type = "AspSection";

            sectionShape.Properties[nameof(RenderAsyncDelegate)] = SectionDelegate(section, sectionShape);

I will do some updates to show what i meant.

@ns8482e i'm still not able to commit on your repo, you didn't add my github account, so i will have to edit the file direcly, but e.g i would have liked to merge the last dev first.

Reply:
>
>         var sectionShape = new Shape();
>         sectionShape.Metadata.Type = "AspSection";
> 
>         sectionShape.Properties[nameof(RenderAsyncDelegate)] = SectionDelegate(section, sectionShape);

@jtkech Nice to learn how to create typed Shape 👍 without using dynamic

Reply:
@ns8482e cool cool !!!